contracts/lib/EIP712.sol

```diff
@@ -57,8 +57,8 @@ library EIP712 {
                     EIP712_DOMAIN_TYPEHASH,
                     keccak256(bytes(name)),
                     keccak256(bytes(version)),
-                    address(this),
-                    bytes32(chainId)
+                    bytes32(chainId),
```

Comment:
Actually came to this repo to open a PR similar to yours 🥹 but hopefully this will be helpful: casting `chainId` to `bytes32` is unnecessary here.